### DISCUSSION TITLE
REST API against local instance of ZkSync

### DISCUSSION THREAD

**Discussion by oac1771** [zkSync-Community-Hub/zksync-developers#56] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Hello, I am trying to hit the /api/v0.2/networkStatus endpoint of a local instance of zksync (https://era.zksync.io/docs/tools/hardhat/testing.html). I am able to hit that endpoint of the main net API with the following curl command: 

`curl -H "content-type: application/json" https://api.zksync.io/api/v0.2/networkStatus` 
and get the following response:

`{"request":{"network":"mainnet","apiVersion":"v02","resource":"/api/v0.2/networkStatus","args":{},"timestamp":"2023-07-11T05:19:00.649764363Z"},"status":"success","error":null,"result":{"lastCommitted":294234,"finalized":294199,"totalTransactions":27605051,"mempoolSize":7,"coreStatus":{"main_database_available":true,"replica_database_available":true,"web3_available":true}}}`

### What is the specific issue or error you're encountering?

When I attempt to make the same request against the local instance of ZkSync I do not see the same behavior and am returned an error.

### Can you share the error messages or logs you're receiving, if any?

`curl -H "content-type: application/json" -X POST http://localhost:3050/api/v0.2/networkStatus`

error:

`{"jsonrpc":"2.0","error":{"code":-32700,"message":"Parse error"},"id":null}`

If I do not include `-X POST` in the curl command I get the following error:

`Used HTTP Method is not allowed. POST or OPTIONS is required`

### Have you made any recent changes to the contract before encountering this issue?

None

### Are there any external libraries or contracts that your contract interacts with?

None

### Can you provide the relevant portions of your contract code where the issue is occurring?

None

### Have you tried to isolate the problem, and if so, what were the results?

None

### What steps have you already taken to try to resolve the issue?

Verified request structure against mainnet api

**Comment by uF4No** [comment, marked as answer]
The docker local node does not expose that API endpoint, only the [web3 RPC API methods](https://era.zksync.io/docs/api/api.html).
Can you explain why do you need it? If it's just to check if the node is running, you can send a simple RPC request like `eth_chainId`.

Also, `https://api.zksync.io/api/v0.2/networkStatus` is the API for zkSync Lite (v1), not Era (v2), and the docker local node is the Era RPC.

**Reply by oac1771** [reply]
Sweet thanks for the reply! I am trying to deploy this to a local kubernetes cluster using minikube and wanted to verify that the zksync setup by testing it post deployment via some api request. Ill look into using the rpc api method for that.

Do you happen to know if the the local node allows for event subscription via websocket (https://docs.zksync.io/dev/events/#setting-up-the-yarn-project)?

**Reply by MexicanAce** [reply]
Yup, the WebSocket endpoint should be exposed on port `3051` and here's the pub/sub documentation for it: https://era.zksync.io/docs/api/api.html#pubsub-api

**Comment by sjewisclare** [comment]
It seems like there might be an issue with the request format or method being used. Double-check the API documentation for the local instance of zkSync to ensure the correct request format and HTTP method are being used. Additionally, verify that the local instance is running and accessible at the specified address and port.

**Comment by tysaboswe** [comment]
It seems like the issue might be related to the request method or formatting. Please ensure that you are using the correct HTTP method (POST) and the request payload is properly formatted according to the API documentation.

**Comment by sinanceviccarter** [comment]
It seems that the local instance of zkSync may have different requirements or restrictions for accessing the /api/v0.2/networkStatus endpoint compared to the mainnet API. Make sure to review the documentation or check with the zkSync team for any specific configuration or setup needed to access this endpoint locally.

**Comment by timmeriweather** [comment]
The issue seems to be related to the incorrect HTTP method used or a parse error in the request to the local instance of zkSync. Ensure that you are using the correct HTTP method (POST) and double-check the request payload for any syntax errors. Additionally, review the documentation or endpoint implementation to ensure compatibility with the request format.

**Comment by estacilo** [comment]
The error message suggests that the issue is with the format of the request being sent to the local instance of zkSync. The -X POST flag is required to specify that the request is a POST request, but it seems that the request is still not being parsed correctly. One possible solution is to ensure that the request body is formatted correctly and includes the necessary parameters for the networkStatus endpoint. Additionally, it may be helpful to check the logs of the local instance of zkSync to see if there are any error messages that provide more information about the issue.
